packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -1167,64 +926,58 @@ class _DialState extends State<_Dial> with SingleTickerProviderStateMixin {
   }
 
   double _getThetaForTime(TimeOfDay time) {
+    final int hoursFactor = widget.use24HourDials ? TimeOfDay.hoursPerDay : TimeOfDay.hoursPerPeriod;
     final double fraction = widget.mode == _TimePickerMode.hour
-      ? (time.hour / TimeOfDay.hoursPerPeriod) % TimeOfDay.hoursPerPeriod
-      : (time.minute / TimeOfDay.minutesPerHour) % TimeOfDay.minutesPerHour;
+        ? (time.hour / hoursFactor) % hoursFactor
+        : (time.minute / TimeOfDay.minutesPerHour) % TimeOfDay.minutesPerHour;
     return (math.pi / 2.0 - fraction * _kTwoPi) % _kTwoPi;
   }
 
-  TimeOfDay _getTimeForTheta(double theta) {
+  TimeOfDay _getTimeForTheta(double theta, {bool roundMinutes = false}) {
     final double fraction = (0.25 - (theta % _kTwoPi) / _kTwoPi) % 1.0;
     if (widget.mode == _TimePickerMode.hour) {
-      int newHour = (fraction * TimeOfDay.hoursPerPeriod).round() % TimeOfDay.hoursPerPeriod;
+      int newHour;
       if (widget.use24HourDials) {
-        if (_activeRing == _DialRing.outer) {
-          if (newHour != 0)
-            newHour = (newHour + TimeOfDay.hoursPerPeriod) % TimeOfDay.hoursPerDay;
-        } else if (newHour == 0) {
-          newHour = TimeOfDay.hoursPerPeriod;
-        }
+        newHour = (fraction * TimeOfDay.hoursPerDay).round() % TimeOfDay.hoursPerDay;
       } else {
+        newHour = (fraction * TimeOfDay.hoursPerPeriod).round() % TimeOfDay.hoursPerPeriod;
         newHour = newHour + widget.selectedTime.periodOffset;
       }
       return widget.selectedTime.replacing(hour: newHour);
     } else {
-      return widget.selectedTime.replacing(
-        minute: (fraction * TimeOfDay.minutesPerHour).round() % TimeOfDay.minutesPerHour
-      );
+      int minute = (fraction * TimeOfDay.minutesPerHour).round() % TimeOfDay.minutesPerHour;
```

Comment:
This might be more readable with an `if else` so that `minute` isn't sometimes being assigned, then reassigned when `roundMinutes` is true